rust/protocol/src/fingerprint.rs

```diff
@@ -136,15 +136,15 @@ impl ScannableFingerprint {
         let same1 = combined
             .local_fingerprint
             .as_ref()
-            .unwrap()
+            .ok_or(SignalProtocolError::FingerprintParsingError)?
```

Comment:
This means you can remove the above `if`. (I like the new form better, honestly.)